### PR TITLE
PAINTROID-478 Flutter: Add pan and zoom canvas support

### PR DESCRIPTION
**Pull request by ikrsk**
Ticket: [https://jira.catrob.at/browse/PAINTROID-478](https://jira.catrob.at/browse/PAINTROID-478)

### CODE REVIEW

File: android/app/src/main/kotlin/org/catrobat/paintroid/MainActivity.kt [hunk 1]
```diff
@@ -24,6 +24,29 @@ class MainActivity : FlutterActivity() {
...
+        setupDeviceChannel(flutterEngine)
+    }
+
+    private fun setupDeviceChannel(flutterEngine: FlutterEngine) {
+        MethodChannel(
+            flutterEngine.dartExecutor.binaryMessenger, "org.catrobat.paintroid/device"
+        ).apply {
+            setMethodCallHandler { call, result ->
+                when (call.method) {
+                    "getHeightInPixels" -> {
```
Comment (juliajulie95) [inline android/app/src/main/kotlin/org/catrobat/paintroid/MainActivity.kt:37]: Could we not use MediaQuery to get the height in flutter?

File: lib/io/src/service/photo_library_service.dart [hunk 2]
```diff
@@ -1,5 +1,3 @@
-import 'dart:typed_data';
```
Comment (juliajulie95) [inline lib/io/src/service/photo_library_service.dart:1]: Do we have a lint check for unused library imports?

File: lib/workspace/src/ui/canvas_painter.dart [hunk 3]
```diff
@@ -4,51 +4,74 @@ import 'package:paintroid/command/command.dart';
...
     );
-    return Stack(
-      fit: StackFit.expand,
-      children: [
-        RepaintBoundary(
-          child: TransparencyGridPattern(
-            numberOfSquaresAlongWidth: 100,
+  }
+
+  Widget get _backgroundLayer {
```
Comment (juliajulie95) [inline lib/workspace/src/ui/canvas_painter.dart:32]: Please do not use helper functions, but classes instead

File: lib/workspace/src/ui/canvas_painter.dart [hunk 4]
```diff
@@ -4,51 +4,74 @@ import 'package:paintroid/command/command.dart';
...
-        ),
-        RepaintBoundary(
-          child: Consumer(
+          );
+        },
+      ),
+    );
+  }
+
+  Widget get _paintingLayer {
```
Comment (juliajulie95) [inline lib/workspace/src/ui/canvas_painter.dart:49]: Please do not use helper functions, but classes instead

File: lib/workspace/src/ui/drawing_canvas.dart [hunk 5]
```diff
@@ -13,40 +15,127 @@ class DrawingCanvas extends ConsumerStatefulWidget {
...
+                data: (_) => _canvasInFittedBox,
+                error: (_) => _canvasInFittedBox,
+                loading: (_) => const SizedBox.shrink(),
+              ),
         ),
       ),
     );
   }
+
+  Widget get _canvasInFittedBox => FittedBox(
```
Comment (juliajulie95) [inline lib/workspace/src/ui/drawing_canvas.dart:137]: Please do not use helper functions, but classes instead

**Comment by ThorstenBandel** [comment]
Finished by https://github.com/Catrobat/Paintroid-Flutter/pull/14

**Pull request closed without merging**